jwst/ami/leastsqnrm.py

```diff
@@ -69,12 +69,12 @@ def weighted_operations(img, model, dqm=None):
     """
     # Remove not-to-be-fit data from the flattened "img" data vector
     flatimg = img.reshape(np.shape(img)[0] * np.shape(img)[1])
-    flatdqm = dqm.reshape(np.shape(img)[0] * np.shape(img)[1])
 
     if dqm is not None:
+        flatdqm = dqm.reshape(np.shape(img)[0] * np.shape(img)[1])
         nanlist = np.where(flatdqm)  # where DO_NOT_USE up.
     else:
-        nanlist = (np.array(()),)  # shouldn't occur w/MAST JWST data
+        nanlist = (np.array((), dtype=np.int32),)  # shouldn't occur w/MAST JWST data
```

Comment:
These changes made it so that `dqm=None` was a valid input to the function, which did not work before